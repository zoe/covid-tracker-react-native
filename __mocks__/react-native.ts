const rn: any = jest.requireActual('react-native');

module.exports = rn;
