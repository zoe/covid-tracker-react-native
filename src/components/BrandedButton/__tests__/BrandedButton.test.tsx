import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, queryByTestId, render } from '@testing-library/react-native';

import { BrandedButton } from '..';

describe('BrandedButton tests', () => {
  it('renders correctly', () => {
    const onPress = jest.fn();
    const tree = renderer.create(<BrandedButton onPress={onPress}>This is a branded button</BrandedButton>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Handles press events', () => {
    const onPress = jest.fn();
    const { getByTestId } = render(<BrandedButton onPress={onPress}>This is a branded button</BrandedButton>);
    expect(onPress).toHaveBeenCalledTimes(0);
    fireEvent.press(getByTestId('buttonTestID'));
    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('Does not fire press events when disabled', () => {
    const onPress = jest.fn();
    const { getByTestId } = render(
      <BrandedButton onPress={onPress} enable={false}>
        This is a branded button
      </BrandedButton>
    );
    expect(onPress).toHaveBeenCalledTimes(0);
    fireEvent.press(getByTestId('buttonTestID'));
    expect(onPress).toHaveBeenCalledTimes(0);
  });
  it('Handles style props being passed', () => {
    const onPress = jest.fn();
    const style = { background: 'red' };
    const { getByTestId } = render(
      <BrandedButton style={style} onPress={onPress} enable={false}>
        This is a branded button
      </BrandedButton>
    );
    expect(getByTestId('buttonTestID').props.style.background).toBe('red');
  });
  it('Shows loading if disabled', () => {
    const onPress = jest.fn();
    const { container } = render(
      <BrandedButton onPress={onPress} enable={false}>
        This is a branded button
      </BrandedButton>
    );
    expect(queryByTestId(container, 'activityIndicator')).toBeTruthy();
  });
  it('Can hide loading when disabled', () => {
    const onPress = jest.fn();
    const { container } = render(
      <BrandedButton onPress={onPress} enable={false} hideLoading>
        This is a branded button
      </BrandedButton>
    );
    expect(queryByTestId(container, 'activityIndicator')).toBeNull();
  });
});
