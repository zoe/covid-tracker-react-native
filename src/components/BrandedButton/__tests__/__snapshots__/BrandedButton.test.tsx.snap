// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandedButton tests renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#024364",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 100,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "borderWidth": undefined,
      "elevation": 0,
      "flexDirection": "row",
      "height": 56,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": undefined,
      "shadowOffset": undefined,
      "shadowOpacity": undefined,
      "shadowRadius": undefined,
    }
  }
  testID="buttonTestID"
>
  <Text
    style={
      Array [
        Object {
          "color": "#565A5C",
          "fontSize": 16,
          "lineHeight": 24,
        },
        Object {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    This is a branded button
  </Text>
</View>
`;
